src/lib/util/SfdxUtils.ts

```diff
@@ -0,0 +1,48 @@
+import childProcess = require('child_process');
+/**
+ * A variable to store SFDX's version number, so we don't have to keep re-running `sfdx-v`
+ */
+let SFDX_VERSION: string;
+
+
+/**
+ * Returns the current version of SFDX installed on the machine, or "unknown" if the version cannot be determined.
+ */
+export async function getSfdxVersion(): Promise<string> {
+	// If we already have a cached value, we can just return that instead of doing anything else.
+	if (SFDX_VERSION !== undefined) {
+		return SFDX_VERSION;
+	}
+
+	// Get the output of `sfdx -v`.
+	let rawVersionString: string;
+	try {
+		rawVersionString = await new Promise<string>((res, rej) => {
+			childProcess.exec('sfdx -v', (err, stdout, stderr) => {
+				if (err) {
+					rej(stderr);
```

Comment:
Ideally, if this fails on an error, there's probably a bigger issue, but just in case, can we somehow log this information or put a dummy sfdx version value and move on? If everything is working for the user and this execution somehow fails, we shouldn't block users from completing the command.